new_client/src/components/events/AddEvent.jsx

```diff
@@ -0,0 +1,250 @@
+      );
+    }
+    return (
+      <button type="submit" className="btn btn-primary">
+        Add Event
+      </button>
+    );
+  }
+
+  componentDidUpdate(nextProps) {
+    if (
+      JSON.stringify(this.props.updatingEvent) !==
+        JSON.stringify(this.state.event) &&
+      JSON.stringify(this.props.updatingEvent) !==
+        JSON.stringify(nextProps.updatingEvent)
+    ) {
+      if (this.props.updatingEvent) {
+        this.setState({ event: this.props.updatingEvent });
+      } else {
+        this.setState({
+          event: {
+            eventName: '',
+            description: '',
+            date: '',
+            venue: '',
+            graduationYear: '',
+            selectedFile: null
+          }
+        });
+      }
+    }
+  }
+
+  render() {
+    return (
+      <div>
+        <div className="jumbotron" style={{ margin: '20px 150px' }}>
+          <form onSubmit={this.handleSubmit} encType="multipart/form-data">
+            <div className="form-group">
+              <label>Event Name:</label>
+              <input
+                type="text"
+                className="form-control"
+                placeholder="Enter event title"
+                name="eventName"
+                value={this.state.event.eventName}
+                onChange={this.handleChange}
+              />
+              <div className="errorMsg">{this.state.error.eventNameError}</div>
+            </div>
+
+            <div className="form-group">
+              <label>Description:</label>
+              <input
+                type="text"
+                className="form-control"
+                placeholder="Enter event description"
+                name="description"
+                value={this.state.event.description}
+                onChange={this.handleChange}
+              />
+              <div className="errorMsg">
+                {this.state.error.descriptionError}
+              </div>
+            </div>
+
+            <div className="form-group">
+              <label>Event Date:</label>
+              <input
+                type="date"
+                className="form-control"
+                name="date"
+                value={this.state.event.date}
+                onChange={this.handleChange}
+              />
+              <div className="errorMsg">{this.state.error.dateError}</div>
+            </div>
+
+            <div className="form-group">
+              <label>Venue:</label>
+              <input
+                type="text"
+                className="form-control"
+                placeholder="Enter event venue"
+                name="venue"
+                value={this.state.event.venue}
+                onChange={this.handleChange}
+              />
+              <div className="errorMsg">{this.state.error.venueError}</div>
+            </div>
+
+            <div className="form-group">
+              <label>Graduation Year:</label>
+              <input
+                type="text"

```

Comment:
Graduation year is a number, so I guess this should have been 
```html
<input
                type="number">
```

Reply:
The type of graduation year in the model is of type string so opting for text was the obvious choice but I'll switch it to number.